File: ILT.IHR.APP/ClientApp/src/app/core/interfaces/Employeew4.ts
export class IEmployeeW4 {
    EmployeeW4ID: number;
    W4TypeID: number;
    W4Type?: string;
    WithHoldingStatusID: number;
    WithHoldingStatus?: string;
    EmployeeID: number;
    EmployeeName: string;
    SSN: string;
    Allowances: number;
    IsMultipleJobsOrSpouseWorks: boolean;
    QualifyingChildren: number;
    OtherDependents: number;
    OtherIncome: string;
    Deductions: string;
    StartDate: Date;
    EndDate: Date;
    CreatedBy: string;
    CreatedDate: Date;
    ModifiedBy: string;
    ModifiedDate: Date;
    TimeStamp: ['']
}
export class IEmployeeW4Display {
    employeeW4ID: number;
    w4TypeID: number;
    w4Type: string;
    WithHoldingStatusID: number;
    withHoldingStatus: string;
    employeeID: number;
    employeeName: string;
    ssn: string;
    allowances: number;
    isMultipleJobsOrSpouseWorks: boolean;
    qualifyingChildren: number;
    otherDependents: number;
    otherIncome: string;
    deductions: string;
    startDate: Date;
    endDate: Date;
    createdBy: string;
    createdDate: Date;
    modifiedBy: string;
    modifiedDate: Date;
}

File: ILT.IHR.DTO/EmployeeW4.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ILT.IHR.DTO
{
    public partial class EmployeeW4 : AbstractDataObject
    {
        public int EmployeeW4ID
        {
            get { return base.RecordID; }
            set { base.RecordID = value; }
        }
        public int EmployeeID { get; set; }
        public string EmployeeName { get; set; }
        [MaxLength(9)]
        [MinLength(9)]
        public string SSN { get; set; }
        [Required]
        [RegularExpression("^0*[1-9]\\d*$")]
        public int W4TypeID { get; set; }
        public string W4Type { get; set; }
        [Required]
        [RegularExpression("^0*[1-9]\\d*$")]
        public int WithHoldingStatusID { get; set; }
        public string WithHoldingStatus { get; set; }
        [RegularExpression("^[0-9]\\d*$")]
        public int? Allowances { get; set; }
        public bool IsMultipleJobsOrSpouseWorks { get; set; }
        [Range(0, 10)]
        public int? QualifyingChildren { get; set; }
        [Range(0, 10)]
        public int? OtherDependents { get; set; }
        [RegularExpression("(\\d+)(\\.)?(\\d+)?")]
        public string OtherIncome { get; set; }
        [RegularExpression("(\\d+)(\\.)?(\\d+)?")]
        public string Deductions { get; set; }
        [Required]
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
    }
}